client/pages/api/auth.ts

```diff
@@ -0,0 +1,54 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { utils } from 'ethers';
+import { supabase } from '../../common/supabase';
+
+function isAddress(address: string) {
+  try {
+    utils.getAddress(address);
+  } catch (e) {
+    console.log(e);
+    return false;
+  }
+  return true;
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { walletAddress } = JSON.parse(req.body);
+
+  if (!isAddress(walletAddress)) {
+    res.status(400).json({ message: 'Please provide a valid address.' });
+    return;
+  }
+
+  // This will be used within a session to verify whether the eth-signature is valid for this nonce.
+  const user = await supabase
+    .from('users')
+    .select()
+    .match({ wallet_address: walletAddress })
```

Comment:
I removed the `wallet_address` attribute in the new schema, we're using `id` instead where I treat that as the wallet address as the primary key